app/jobs/process_attachment_job.rb

```diff
@@ -0,0 +1,17 @@
+require "attachment_processors/image"
+
+class ProcessAttachmentJob < ActiveJob::Base
+  queue_as :default
+
+  def perform(attachment)
+    if attachment.image?
+      processor = AttachmentProcessor::Image.new(attachment)
+
+      attachment.thumbnail_sizes.each do |tag, dimensions|
```

Comment:
Thinking this logic should be a part of `AttachmentProcessor::Image`, we could then do something more automatic in here:

```rb
def perform(attachment)
  processor_for(attachment).run
end

private

def processor_for(attachment)
  case attachment.file_type
  when %r{\Aimage/}
    AttachmentProcessor::Image
  else
    AttachmentProcessor::NullProcessor
  end
end
```